chain/network/src/peer_manager/peer_store.rs

```diff
@@ -56,56 +56,82 @@ impl PeerStore {
         store: Arc<Store>,
         boot_nodes: &[PeerInfo],
     ) -> Result<Self, Box<dyn std::error::Error>> {
-        let mut peer_states = HashMap::default();
-        let mut addr_peers = HashMap::default();
-
-        for peer_info in boot_nodes.iter() {
-            if !peer_states.contains_key(&peer_info.id) {
+        // A mapping from `PeerId` to `KnownPeerState`.
+        let mut peerid_2_state = HashMap::default();
+        // Stores mapping from `SocketAddr` to `VerifiedPeer`, which contains `PeerId`.
+        // Only one peer can exist with given `PeerId` or `SocketAddr`.
+        // In case of collision, we will choose the first one.
+        let mut addr_2_peer = HashMap::default();
+
+        let now = Clock::utc();
+        boot_nodes.iter().for_each(|peer_info|{
+            if !peerid_2_state.contains_key(&peer_info.id) {
                 if let Some(peer_addr) = peer_info.addr {
-                    match addr_peers.entry(peer_addr) {
+                    match addr_2_peer.entry(peer_addr) {
                         Entry::Occupied(entry) => {
                             // There is already a different peer_id with this address.
                             error!(target: "network", "Two boot nodes have the same address {:?}", entry.key());
+                            // TODO: add panic!, `boot_nodes` list is wrong
                         }
                         Entry::Vacant(entry) => {
                             entry.insert(VerifiedPeer::signed(peer_info.id.clone()));
-                            peer_states.insert(
+                            peerid_2_state.insert(
                                 peer_info.id.clone(),
-                                KnownPeerState::new(peer_info.clone()),
+                                KnownPeerState::new(peer_info.clone(), now),
                             );
                         }
                     }
                 }
             }
-        }
+            else {
+                // TODO: add panic!, `boot_nodes` list is wrolng
```

Comment:
typo in wrong.

Any reason why we don't add panic right now? 

(or at least an error message)